documentation-site/components/yard/config/file-uploader.ts

```diff
@@ -0,0 +1,102 @@
+import {FileUploader} from 'baseui/file-uploader';
+import {PropTypes} from '../const';
+import {TConfig} from '../types';
+
+const changleHandlers = [
+  'onClick',
+  'onFocus',
+  'onBlur',
+  'onKeyDown',
+  'onDragStart',
+  'onDragEnter',
+  'onDragOver',
+  'onDragLeave',
+  'onDrop',
+  'onDropAccepted',
+  'onDropRejected',
+  'onFileDialogCancel',
+  'onCancel',
+  'onRetry',
+];
+
+const FileUploaderConfig: TConfig = {
+  imports: {
+    'baseui/file-uploader': {named: ['FileUploader']},
+  },
+  scope: {
+    FileUploader,
+  },
+  theme: [],
+  props: {
+    accept: {
+      value: '',
+      type: PropTypes.String,
+      description:
+        'Set accepted file types. See https://github.com/okonet/attr-accept for more information',
+    },
+    maxSize: {
+      value: undefined,
+      type: PropTypes.Number,
+      description: 'Maximum file size (in bytes).',
+    },
+    minSize: {
+      value: undefined,
+      type: PropTypes.Number,
+      description: 'Minimum file size (in bytes).',
+    },
+    multiple: {
+      value: undefined,
+      type: PropTypes.Boolean,
+      description:
+        'Allow drag n drop (or selection from the file dialog) of multiple files',
+    },
+    disableClick: {
+      value: undefined,
+      type: PropTypes.Boolean,
+      description:
+        'Disallow clicking on the dropzone container to open file dialog.',
+    },
+    disabled: {
+      value: false,
+      type: PropTypes.Boolean,
+      description: 'Renders component in disabled state.',
+    },
+    errorMessage: {
+      value: '',
+      type: PropTypes.String,
+      description: 'Error message to be displayed.',
+      stateful: true,
+    },
+    ...changleHandlers.reduce((acc, current) => {
```

Comment:
I wonder if we should come up with some more robust mechanism to share parts of the documentation. Maybe having `config/common.ts` with all these handlers and then just picking them across other config files. We already do a lot of sharing for input components. Wdyt?

Reply:
that makes sense - I'll do that as part of this PR